### PR TITLE
FALCON-2267 Definition support for custom extensions

### PR DESCRIPTION
**Pull request by sandeepSamudrala**

### CODE REVIEW

**Comment by sandeepSamudrala** [comment]
Dev-Tested 
[sandeep@/usr/local/falcon/falcon-distributed-0.11-SNAPSHOT$]falcon extension -definition -extensionName extension-example
"testExtensionDesc"
[sandeep@/usr/local/falcon/falcon-distributed-0.11-SNAPSHOT$]

**Comment by PraveenAdlakha** [comment]
Please paste the exact commands you used to dev test this one.

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java [hunk 1]
```diff
@@ -693,9 +693,17 @@ public APIResult registerExtensionMetadata(
...
         try {
-            return new APIResult(APIResult.Status.SUCCEEDED, ExtensionStore.get().getResource(extensionName,
-                    extensionName.toLowerCase() + EXTENSION_PROPERTY_JSON_SUFFIX));
+            ExtensionType extensionType = getExtensionType(extensionName);
+            if (ExtensionType.TRUSTED.equals(extensionType)) {
+                return new APIResult(APIResult.Status.SUCCEEDED, ExtensionStore.get().getResource(extensionName,
+                        extensionName.toLowerCase() + EXTENSION_PROPERTY_JSON_SUFFIX));
+            } else {
+                return new APIResult(APIResult.Status.SUCCEEDED,
+                        ExtensionStore.getMetaStore().getDetail(extensionName).getDescription());
```
Comment (pallavi-rao) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: This is returning the short description which the user supplied at the time of registration. What should be returned is the content of file/files under META.

**Comment by sandeepSamudrala** [comment]
Closing this pull request as the same is taken care in another pull request.

**Pull request closed without merging**